### ISSUE TITLE
Row index not matching selected index unable to update row

### ISSUE THREAD

**Issue by minsis** [wenzhixin/bootstrap-table#5319]
**Description**  
I have a function to update a single row on the backend, however I'm unable to update it on the frontend because the row index doesn't match up with the selected index numbers from the database.

using $table.bootstrapTable('getSelections') gets me the row data and still not the index data. So I'm not really sure how to get the index of the row itself without it matching the actual database id column.

Sorry if this is a dumb question but I'm not very good with javascript at this point and am only getting by =)

**Comment by minsis** [comment]
I should also mention looked at getRowByUniqueId but it doesn't seem to return anything for me even if I use something like index 1 hard coded or using the database ID

**Comment by UtechtDustin** [comment]
The method `getSelections` returns all checked rows, which means if you row data contains the database index you easily can fetich it like

```
const selection = $table.bootstrapTable('getSelections')
console.log(selection[0]['databaseId']);
```
This example would return the database id of the first row which is checked.

> I should also mention looked at getRowByUniqueId but it doesn't seem to return anything for me even if I use something like index 1 hard coded or using the database ID

It should work fine (check the example), i guess you didnt defined the unique id field (which should be your database id).
https://live.bootstrap-table.com/example/methods/get-row-by-unique-id.html
-------

If you need more help please provide us an example using our [edito](https://live.bootstrap-table.com/).

**Comment by minsis** [comment]
So the issue with getRowByUniqueId was not being able to update selected row using the same ID because I had no idea what the row index number was. However, I found updateByUniqueId today which is what I needed to update the row in-place using the database ID.

Sorry for the noob stuff, I'm really new to javascript and I'm barely getting by what what I need to get my project done. I appreciate your help!
